tests/nested/manual/uc20-install-in-initrd/task.yaml

```diff
@@ -69,26 +70,32 @@ restore: |
   rm -rf ~/.snap/gnupg
 
 execute: |
-  # TODO: in a normal run, ubuntu-data-<someid> is used instead of
-  # just "ubuntu-data". We need to figure out if this is OK.
-  remote.exec "ls /dev/mapper/ubuntu-data*"
-  remote.exec "ls /dev/mapper/ubuntu-save*"
   remote.exec "cat /proc/cmdline" | MATCH "snapd_recovery_mode=install"
-
   remote.exec "cat /var/lib/snapd/modeenv" > modeenv
 
   MATCH "mode=run" <modeenv
 
-  boot_grub_sha3="$(remote.exec "cat /run/mnt/ubuntu-boot/EFI/boot/grubx64.efi" | "$TESTSLIB"/tools/sha3-384)"
-  seed_grub_sha3="$(remote.exec "cat /run/mnt/ubuntu-seed/EFI/boot/grubx64.efi" | "$TESTSLIB"/tools/sha3-384)"
-  seed_shim_sha3="$(remote.exec "cat /run/mnt/ubuntu-seed/EFI/boot/bootx64.efi" | "$TESTSLIB"/tools/sha3-384)"
+  if [ "${INSTALL_MODE}" != none ]; then
+    # TODO: in a normal run, ubuntu-data-<someid> is used instead of
+    # just "ubuntu-data". We need to figure out if this is OK.
+    remote.exec "ls /dev/mapper/ubuntu-data*"
+    remote.exec "ls /dev/mapper/ubuntu-save*"
 
-  boot_grub_asset_sha3="$(remote.exec "cat /var/lib/snapd/boot-assets/grub/grubx64.efi-${boot_grub_sha3}" | "$TESTSLIB"/tools/sha3-384)"
-  seed_grub_asset_sha3="$(remote.exec "cat /var/lib/snapd/boot-assets/grub/grubx64.efi-${seed_grub_sha3}" | "$TESTSLIB"/tools/sha3-384)"
-  seed_shim_asset_sha3="$(remote.exec "cat /var/lib/snapd/boot-assets/grub/bootx64.efi-${seed_shim_sha3}" | "$TESTSLIB"/tools/sha3-384)"
+    boot_grub_sha3="$(remote.exec "cat /run/mnt/ubuntu-boot/EFI/boot/grubx64.efi" | "$TESTSLIB"/tools/sha3-384)"
+    seed_grub_sha3="$(remote.exec "cat /run/mnt/ubuntu-seed/EFI/boot/grubx64.efi" | "$TESTSLIB"/tools/sha3-384)"
+    seed_shim_sha3="$(remote.exec "cat /run/mnt/ubuntu-seed/EFI/boot/bootx64.efi" | "$TESTSLIB"/tools/sha3-384)"
 
-  [ "${boot_grub_sha3}" = "${boot_grub_asset_sha3}" ]
-  [ "${seed_grub_sha3}" = "${seed_grub_asset_sha3}" ]
-  [ "${seed_shim_sha3}" = "${seed_shim_asset_sha3}" ]
+    boot_grub_asset_sha3="$(remote.exec "cat /var/lib/snapd/boot-assets/grub/grubx64.efi-${boot_grub_sha3}" | "$TESTSLIB"/tools/sha3-384)"
+    seed_grub_asset_sha3="$(remote.exec "cat /var/lib/snapd/boot-assets/grub/grubx64.efi-${seed_grub_sha3}" | "$TESTSLIB"/tools/sha3-384)"
+    seed_shim_asset_sha3="$(remote.exec "cat /var/lib/snapd/boot-assets/grub/bootx64.efi-${seed_shim_sha3}" | "$TESTSLIB"/tools/sha3-384)"
 
-  # TODO: When testing with NESTED_ENABLE_TPM=true, check /var/lib/snapd/device/fde/boot-chains
+    [ "${boot_grub_sha3}" = "${boot_grub_asset_sha3}" ]
+    [ "${seed_grub_sha3}" = "${seed_grub_asset_sha3}" ]
+    [ "${seed_shim_sha3}" = "${seed_shim_asset_sha3}" ]
+  fi
+
+  if [ "${INSTALL_MODE}" = secureboot ]; then
+    remote.exec test -f /var/lib/snapd/device/fde/boot-chains
+  else
+    remote.exec ! test -f /var/lib/snapd/device/fde/boot-chains
```

Comment:
better `remote.exec not ...` (`set -e` and `!` do not interact as expected, although here it might be fine as the command is actually remote.exec, but still better be consistent).